### ISSUE TITLE
Not reading sysvol or netlogon shares

### ISSUE THREAD

**Issue by hkelley** [SnaffCon/Snaffler#31]
Config:

```
ComputerTargets = ["XXXX"]
ScanSysvol = true
ScanNetlogon = true
```

Output

```
2021-03-07 22:40:05 -05:00 [Info] Creating a sharefinder task for XXXXXX
2021-03-07 22:40:05 -05:00 [Info] Created all sharefinder tasks.
 . . . .
 
2021-03-07 22:40:35 -05:00 [Info] Status Update:
ShareFinder Tasks Completed: 1
ShareFinder Tasks Remaining: 0
ShareFinder Tasks Running: 0
TreeWalker Tasks Completed: 0
TreeWalker Tasks Remaining: 0
TreeWalker Tasks Running: 0
FileScanner Tasks Completed: 0
FileScanner Tasks Remaining: 0
FileScanner Tasks Running: 0
```

No SYSVOL or NETLOGON scanned.

**Comment by hkelley** [comment]
I debugged this briefly.   It looks like these shares are matching a classifier and therfore are being skipped for TreeWalker creation.

From ShareFinder.cs
```
                   // classify them
                    foreach (ClassifierRule classifier in MyOptions.ShareClassifiers)
                    {
                        ShareClassifier shareClassifier = new ShareClassifier(classifier);
                        if (shareClassifier.ClassifyShare(shareName))
                        {
                            matched = true;
                            break;
                        }
                    }
                    // by default all shares should go on to TreeWalker unless the classifier pulls them out.
                    // send them to TreeWalker
                    if (!matched)
                    {
```

**Comment by l0ss** [comment]
You're right that doesn't make sense. There's some special logic in place around sysvol/netlogon so maybe if you're specifying computer targets that's skipping over it somehow?

Just to make sure... your computer targets include a DC and the account you're running as has read access to the sysvol share on it?

**Comment by hkelley** [comment]
I am running a single computer (DC) test job and our environment is typical,  all users can read SYSVOL.

I'm still trying two wrap my head around the classifier assignment logic but the summary of the issue seems to be that the **matched** variable is being set to true after iterating over the ClassifierRules, and therefore skipping the TreeWalker task assignment.

I will try to drill into the KeepShareBlack evaluation to understand why this rule it hitting for those two shares.

**Comment by hkelley** [comment]
Again,  I don't pretend to fully understand the classifier rules,  but this might be a solution.   I moved the SYSVOL and NETLOGON logic out of ShareClassifier.cs and up into ShareFindinder.cs.  

I am assuming that if a user sets ScanSysvol  then that should override any classifier rules.

```
            foreach (HostShareInfo hostShareInfo in hostShareInfos)
            {
                string shareName = GetShareName(hostShareInfo, computer);
                if (!String.IsNullOrWhiteSpace(shareName))
                {
                    bool matched = false;

                    //HCK -  move the sysvol and netlogon logic out/above the classifier rules

                    // first time we hit sysvol, 
                    switch (hostShareInfo.shi1_netname.ToLower())
                    {    
                        case "sysvol":
                            if (MyOptions.ScanSysvol == true)
                            {
                                //  leave matched as false so that we don't suppress the TreeWalk
                                //  toggle the flag so that ever other share replica will be skipped
                                MyOptions.ScanSysvol = false;
                                break;
                            }
                            matched = true;
                            break;
                        case "netlogon":
                            if (MyOptions.ScanNetlogon == true)
                            {
                                
                                //  leave matched as false so that we don't suppress the TreeWalk
                                //  toggle the flag so that ever other share replica will be skipped
                                MyOptions.ScanNetlogon = false;
                                break;
                            }
                            matched = true;
                            break;
                        default:
                            // classify them
                            foreach (ClassifierRule classifier in MyOptions.ShareClassifiers)
                            {
                                ShareClassifier shareClassifier = new ShareClassifier(classifier);
                                if (shareClassifier.ClassifyShare(shareName))
                                {
                                    // HCK   ugly hack
                                    System.Console.WriteLine(shareName + " " + classifier.RuleName);

                                    matched = true;
                                    break;
                                }
                            }
                            break;
                    }
```

**Comment by hkelley** [comment]
PR #32 with proposed change to correct this

**Comment by l0ss** [comment]
fixed in PR #32
